src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py

```diff
@@ -16,8 +16,8 @@
 register_cli_argument('ad app', 'identifier', options_list=('--id',), help='identifier uri, application id, or object id')
 register_cli_argument('ad app', 'identifier_uris', nargs='+', help='space separated unique URIs that Azure AD can use for this app.')
 register_cli_argument('ad app', 'reply_urls', nargs='+', help='space separated URIs to which Azure AD will redirect in response to an OAuth 2.0 request. The value does not need to be a physical endpoint, but must be a valid URI.')
-register_cli_argument('ad app', 'start_date', help='the start date after which password or key would be valid. Default value is current time')
-register_cli_argument('ad app', 'end_date', help='the end date till which password or key is valid. Default value is one year after current time')
+register_cli_argument('ad app', 'start_date', help="the date in iso8601 format when credentials becoming valid, e.g. '2017-01-01T01:00:00+00:00' or '2017-01-01'. Default value is current time")
+register_cli_argument('ad app', 'end_date', help="the date in iso8601 format when credentials get expired, e.g. '2017-12-31T11:59:59+00:00' or '2017-12-31'. Default value is one year after current time")
```

Comment:
Some recommended wording changes:
"--start-date: Date or datetime at which credentials become valid. (e.g. ....)"
"--end-date: Date or datetime after which credentials expire. (e.g. ...)"

I would recommend putting the stuff about format and the examples of using the format into the long-summary for the parameters, which means they would need to be moved to the help.py file.